app.rb

```diff
@@ -0,0 +1,106 @@
+require_relative 'lib/database_connection'
+require_relative "lib/PostsRepository"
+require_relative 'lib/UsersRepository'
+require 'sinatra/reloader'
+require 'sinatra'
+
+DatabaseConnection.connect
+
+class Application < Sinatra::Base
+  # This allows the app code to refresh
+  # without having to restart the server.
+  configure :development do
+    register Sinatra::Reloader
+    also_reload 'lib/album_repository'
+    also_reload 'lib/artist_repository'
+    enable :sessions
+  end
+
+  get '/' do 
+    return erb(:home_page)
+  end 
+
+  get '/stream' do
+     repo = PostsRepository.new
```

Comment:
Do you need to call __Repository.new every time? Did you experiment with calling it once possibly as a session variable?